magenta-lib/src/main/scala/magenta/json/JValueExtractable.scala

```diff
@@ -29,14 +28,16 @@ object JValueExtractable {
     def extract(json: JValue) = extractOption(JArray.unapply)(json) map (_.flatMap(jve.extract(_)))
   }
   implicit def MapExtractable[V](implicit jve: JValueExtractable[V]) = new JValueExtractable[Map[String,V]] {
-    def extract(json: JValue) = {
-      val kvs: List[(String, Option[V])] = for {
-        JObject(fields) <- json
+    def extract(json: JValue): Option[Map[String, V]] = {
+      val JObject(fields) = json
```

Comment:
Note, this has been lifted out of the for expression to fix a nasty bug when extracting nested maps. Previously, it would fail, I think because JObject(fields) <- json would act recursively, which caused problems later on. Very subtle and tricky to work out what was going on.